src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1578,6 +1579,33 @@ public void SaveAs(string path, bool isBackup = false)
             }
         }
 
+        /// <summary>
+        /// Save the current workspace to a specific file path. If the file path is null or empty, an
+        /// exception is written to the console.
+        /// </summary>
+        /// <param name="path">The path to the file.</param>
+        /// <param name="saveContext">The context of the save operation.</param>
+        /// <param name="isBackup">Indicates if an automated backup save has called this function.</param>
+        /// <exception cref="IOException"></exception>
+        /// <exception cref="UnauthorizedAccessException"></exception>
+        public void SaveAs(string path, SaveContext saveContext, bool isBackup = false)
+        {
+            try
+            {
+                Model.Logger.Log(String.Format(Properties.Resources.SavingInProgress, path));
```

Comment:
it seems one of these methods could just call the other instead of duplicating code, or use a shared private method.